lib/main.ts

```diff
@@ -80,19 +110,51 @@ export class Transpiler {
     this.result = '';
     this.errors = [];
     this.lastCommentIdx = -1;
-    this.currentFile = sourceFile.getSourceFile();
+    this.currentFile = sourceFile;
+    if (this.generateSourceMap) {
+      this.sourceMap = new SourceMap.SourceMapGenerator({file: this.relativeFileName + '.dart'});
+      this.sourceMap.setSourceContent(this.relativeFileName, this.currentFile.text);
+    }
     this.visit(sourceFile);
     if (this.errors.length) {
       var e = new Error(this.errors.join('\n'));
       e.name = 'TS2DartError';
       throw e;
     }
+
     return this.result;
   }
 
+  enterNode(n: ts.Node) {
+    if (!this.sourceMap) return; // source maps disabled.
```

Comment:
if this method is only for sourcemap generation, maybe it should have a different name?